examples/DropboxAndroid/README.md

```diff
@@ -0,0 +1,112 @@
+```
+val scopes = listOf("account_info.read", "files.content.write")
+Auth.startOAuth2PKCE(this, APP_KEY, requestConfig, scopes)
+```
+5. When the user has completed the authentication flow they will be taken back to your app. In `onResume` you can call `Auth.getDbxCredential()` to get the resulting `DbxCredential`.
+6. To make requests using the Dropbox Java SDK, create a `DbxClientV2` like so. ClientIdentifiers are usually of the form "SoftwareName/SoftwareVersion". `DbxClientV2` will have the request objects needed to make a call to the Dropbox API.
+```
+val clientIdentifier = "DropboxSampleAndroid/1.0.0"
+val requestConfig = DbxRequestConfig(clientIdentifier)
+val dropboxClient = DbxClientV2(requestConfig, credential)
+```
+The Dropbox Java SDK will use the long lived refresh token found in the DbxCredential to automatically refresh your short lived access token when it makes requests.
+
+### Hiding your Dropbox App Secret
+
+If you want to hide your Dropbox app secret, you can follow these steps:
+1. Edit your `local.properties` file to contain the following line with your Dropbox key from the developer console
+```
+DROPBOX_KEY=mydropboxkeygoeshere
+```
+2. Edit your `build.gradle` to include the following:
+```
+android {
+    defaultConfig {
+        //more config
+
+        Properties localProperties = getLocalProperties()
+        String dropboxKey = localProperties['DROPBOX_KEY']
+        buildConfigField "String", "DROPBOX_KEY", "\"${dropboxKey}\""
+        manifestPlaceholders = [dropboxKey: dropboxKey]
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+}
+
+def getLocalProperties() {
+    Properties props = new Properties()
+    if (file('../local.properties').exists()) {
+        props.load(new FileInputStream(file('../local.properties')))
+    }
+    return props
+}
+```
+3. Edit your `AndroidManifest` to use the manifest placeholder defined in step 2.
+```
+    <activity
+        android:name="com.dropbox.core.android.AuthActivity"
+        android:configChanges="orientation|keyboard"
+        android:launchMode="singleTask">
+        <intent-filter>
+            <!-- Using a Manifest Placeholder to hide our app key -->
+            <data android:scheme="db-${dropboxKey}" />
+            <action android:name="android.intent.action.VIEW" />
+            <category android:name="android.intent.category.BROWSABLE" />
+            <category android:name="android.intent.category.DEFAULT" />
+        </intent-filter>
+    </activity>
+```
+4. Use your app key with:
+```
+val APP_KEY = BuildConfig.DROPBOX_KEY
+```
+
+## Authentication with the Dropbox Java SDK
+Starting September 30, 2021 the Dropbox authentication flow will only return short lived tokens. The Java SDK has been updated to support this using `DbxCredential`s.
```

Comment:
Let's add a link to https://dropbox.tech/developers/migrating-app-permissions-and-access-tokens somewhere in here for context.